frontend/src/app/general/error-handling/error-handling.interceptor.ts

```diff
@@ -57,28 +55,29 @@ export class ErrorHandlingInterceptor implements HttpInterceptor {
           }
         },
         error: (err) => {
-          if (err.error.detail.err_code == 'token_exp') {
+          if (err.error.detail?.err_code == 'token_exp') {
             return;
           }
 
           if (
             typeof err.error !== 'undefined' &&
             typeof err.error.detail !== 'undefined'
           ) {
-            if (Array.isArray(err.error.detail)) {
+            let detail = err.error.detail;
+            if (Array.isArray(detail)) {
               // Pydantic errors
-              for (let error of err.error.detail) {
+              for (let error of detail) {
                 this.toastService.showError(
                   getReasonPhrase(err.status),
                   error.msg
                 );
               }
-            } else if (err.error.detail.reason) {
+            } else if (detail.reason) {
+              if (detail.reason && Array.isArray(detail.reason)) {
+                detail.reason = detail.reason.join(' ');
+              }
```

Comment:
There is another statement below the if statement, so this doesn't work :(

Reply:
```suggestion
            } else if (detail.reason) {
              if (Array.isArray(detail.reason)) {
                detail.reason = detail.reason.join(' ');
              }
```
Good point, I didn't see it. But then I think the `detail.reason` in the inner if can be removed since it is already checked in the outer else if.